qhana_plugin_runner/tasks.py

```diff
@@ -31,6 +81,9 @@ def save_task_result(self, task_log: str, db_id: int):
         # TODO use better fitting error
         raise KeyError(f"Could not find db entry for id {db_id}, saving task log failed!")
 
+    if task_data.multi_step:
+        task_data.progress_value = task_data.progress_target
```

Comment:
if the plugin does not provide any progress then we should not try to provide any progress. Besides progress value for a finished task is pretty pointless. (although it may not be a bad idea to always set the progress to the target value here)